### PR TITLE
[#479] Remove redundadnt k8s properties changes

### LINKED ISSUE #479: Redundant ConfigMap update events in callback mechanism
When legion.k8s.watch is restarted, it generates new event about ConfigMap update even if ConfigMap hasn't been modified.

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This closes #479 
This filters k8s events that script gets on start and on watcher restarts

### CODE REVIEW

**Comment by kirillmakhonin** [comment]
test this please

**Comment by kirillmakhonin** [comment]
Test this please

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test all

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by alexander-semenets** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

File: deploy/helms/legion/templates/metrics/grafana.yaml [hunk 1]
```diff
@@ -35,6 +35,8 @@ spec:
           value: "true"
         - name: GF_INSTALL_PLUGINS
           value: ""
+        - name: GF_LOG_LEVEL
+          value: "debug"
```
Comment (kirillmakhonin) [inline deploy/helms/legion/templates/metrics/grafana.yaml]: Add log level Profile parameter

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test all

**Comment by kirillmakhonin** [comment]
bot, test all

File: legion/legion/k8s/properties.py [hunk 2]
```diff
@@ -655,12 +671,15 @@ def _read_data_from_dict(self, source_dict):
...
-        :return: None
+        :return: bool -- is new data has been received
         """
-        self._state = {
+        new_data = {
             k: base64.b64decode(v.encode('ascii')).decode('utf-8')
             for k, v in source_dict.items()
         }
+        data_has_been_changed = self._state != new_data
+        self._state = new_data
```
Comment (dsuslov) [inline legion/legion/k8s/properties.py]: @kirillmakhonin what is the reason to self._state = new_data  if  data_has_been_changed = false ?

File: deploy/helms/legion/templates/metrics/grafana.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kirillmakhonin) [reply deploy/helms/legion/templates/metrics/grafana.yaml]: outdated comment

**Comment by kirillmakhonin** [comment]
@dsuslov please re-review
Tests will be started when develop branch became working

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test all

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Comment by kirillmakhonin** [comment]
bot, test app

**Pull request merged**